File: backend/app/models/seminar.py
from sqlalchemy import Column, String, Text, Integer, DateTime, ForeignKey
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
import uuid
from ..db.base import Base


class Seminar(Base):
    __tablename__ = "seminars"
    
    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    title = Column(String(255), nullable=False)
    description = Column(Text)
    scheduled_date = Column(DateTime(timezone=True), nullable=False, index=True)
    duration_minutes = Column(Integer, default=90)
    location = Column(String(255))
    meeting_link = Column(String(500))
    status = Column(String(50), default='SCHEDULED', index=True)
    created_by = Column(UUID(as_uuid=True), ForeignKey('users.id'))
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())

File: frontend/src/types/seminar.ts
export interface Seminar {
  id: string
  title: string
  description?: string
  scheduled_date: string
  duration_minutes?: number
  location?: string
  meeting_link?: string
  status?: string
  created_at?: string
  // Workflow assignment fields (optional; set by coordinator during schedule creation)
  presenter_id?: string
  technical_moderator_id?: string
}
